impress-org/givewp-next-gen
# Feature: Add custom field support in Donation Details, Donor Details

## Description



This PR adds support for displaying custom fields/values in the Donation Details and Donor Details screens in the WordPress/GiveWP admin dashboard.

## Affects



- Adds a "Display Settings" section to custom fields.
- Adds the "Display in Admin" toggle in the Display Settings section.
- Adds Controller/View for the Donation Details metabox.
- Adds Controller/View for the Donor Details table.
- Adds `shouldDisplayInAdmin` macro to `Give\Framework\FieldsAPI\Field`.

## Visuals



### Display Settings inspector section
![Display Settings inspector section](https://user-images.githubusercontent.com/10858303/205311825-b09901c2-db6b-4d6e-87a3-6b7a7f60e38e.png)

### Donation Details metabox
![Donation Details metabox](https://user-images.githubusercontent.com/10858303/205311822-3c23b9dc-b71c-4ba9-b50e-73be718a52c7.png)

### Donor Details table
![Donor Details table](https://user-images.githubusercontent.com/10858303/205311816-0d36fd5e-4b24-4bbd-976d-1ec1921e508e.png)

## Testing Instructions



- Add a custom field
- Configure to "Display in Admin"
- Submit the donation form.
- Custom field label/value should be displayed in the Donation Details screen.
- Repeat (using "Store as Donor Meta" setting) for Donor custom field.
